kitsune/search/es_utils.py

```diff
@@ -886,3 +890,52 @@ def indexes_for_doctypes(doctype):
     # Import locally to avoid circular import.
     from kitsune.search.models import get_mapping_types
     return set(d.get_index() for d in get_mapping_types(doctype))
+
+
+def handle_es_errors(template, status_code=503):
+    """Handles Elasticsearch exceptions for views
+
+    Wrap the entire view in this and don't worry about Elasticsearch exceptions
+    again!
+
+    :arg template: template path string or function to generate the template
+        path string for HTML requests
+    :arg status_code: status code to return
+
+    :returns: content-type-appropriate HttpResponse
+
+    """
+    def handler(fun):
+        @wraps(fun)
+        def _handler(request, *args, **kwargs):
+            try:
+                return fun(request, *args, **kwargs)
+
+            except ES_EXCEPTIONS as exc:
+                is_json = (request.GET.get('format') == 'json')
+                callback = request.GET.get('callback', '').strip()
+                content_type = 'application/x-javascript' if callback else 'application/json'
+                if is_json:
+                    return HttpResponse(
+                        json.dumps({'error': _('Search Unavailable')}),
+                        content_type=content_type,
+                        status=status_code)
+
+                # If template is a function, call it with the request, args
+                # and kwargs to get the template.
+                if callable(template):
+                    template = template(request, *args, **kwargs)
+
+                # FIXME: statsd exceptions
+                # # Cheating here: Convert from 'Timeout()' to 'timeout' so
+                # # we have less code, but still have good stats.
+                # exc_bucket = repr(exc).lower().strip('()')
+                # statsd.incr('search.esunified.{0}'.format(exc_bucket))
+
+                # FIXME: Log exceptions
+                # log.exception(exc)
```

Comment:
I can reimplement these, but I'm not sure we care. I know I haven't looked at them in ages. Do we care? If not, then let's nix them. We can always re-add them later.